### PR TITLE
Migrated UIManagerModuleConstantsTest.java to UIManagerModuleConstant…

### PR DESCRIPTION
**Pull request by giolaq**
## Summary:

This PR converts UIManagerModuleConstantsTest.java into Kotlin as requested in [https://github.com/facebook/react-native/issues/38825](https://github.com/facebook/react-native/issues/38825)

## Changelog:

[INTERNAL] [CHANGED] - Convert to UIManagerModuleConstantsTest.java to UIManagerModuleConstantsText.kt

## Test Plan:

```./gradlew :packages:react-native:ReactAndroid:test``` must pass and CI must be green

Screenshot 2023-08-14 at 17 43 14

### CODE REVIEW

**Comment by giolaq** [comment]
PR is ready for review @cortinico @mdvacca. 3 circleci have unexpected errors that should not be related to code

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+import org.powermock.core.classloader.annotations.PowerMockIgnore
+import org.powermock.modules.junit4.rule.PowerMockRule
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+@RunWith(RobolectricTestRunner::class)
+@PowerMockIgnore("org.mockito.*", "org.robolectric.*", "androidx.*", "android.*")
+class UIManagerModuleConstantsTest {
+
+  @get:Rule var rule = PowerMockRule()
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:27]:
```suggestion
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+  private class ConcreteViewManager(val viewName: String) : SimpleViewManager<View?>() {
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:29]:
```suggestion
  private class ConcreteViewManager(val viewName: String) : SimpleViewManager<View>() {
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+    fun setExportedCustomDirectEventTypeConstants(
+      constants: MutableMap<String, Map<String, Any>?>
+    ) {
+      customDirectEventTypeConstants = constants
+    }
+
+    override fun createViewInstance(reactContext: ThemedReactContext): View {
+      return View(reactContext)
+    }
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:42]:
```suggestion
    override fun createViewInstance(reactContext: ThemedReactContext): View = View(reactContext)
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+
+  @get:Rule var rule = PowerMockRule()
+
+  private class ConcreteViewManager(val viewName: String) : SimpleViewManager<View?>() {
+
+    private var customDirectEventTypeConstants: MutableMap<String, Map<String, Any>?> =
+      mutableMapOf()
+
+    fun setExportedCustomDirectEventTypeConstants(
+      constants: MutableMap<String, Map<String, Any>?>
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:35]:
```suggestion
      constants: MutableMap<String, Map<String, Any>>
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+      customDirectEventTypeConstants = constants
+    }
+
+    override fun createViewInstance(reactContext: ThemedReactContext): View {
+      return View(reactContext)
+    }
+
+    override fun getName(): String {
+      return viewName
+    }
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:46]:
```suggestion
    override fun getName(): String = viewName
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+
+    private val TWIRL_DIRECT_EVENT_MAP: Map<String, Any>? =
+      MapBuilder.of("registrationName", "onTwirl")
+
+    private const val VIEW_MANAGER_NAME = "viewManagerName"
+    private const val BUBBLING_EVENTS_TYPES_KEY = "bubblingEventTypes"
+    private const val DIRECT_EVENTS_TYPES_KEY = "directEventTypes"
+
+    @Suppress("UNCHECKED_CAST")
+    private fun valueAtPath(nestedMap: Map<String, Any>?, vararg keyPath: String): Any? {
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:234]:
```suggestion
    private fun valueAtPath(nestedMap: Map<String, Any>, vararg keyPath: String): Any? {
```

File: packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+      var nestedMap = nestedMap
+      Assertions.assertThat(keyPath).isNotEmpty
+      var value: Any? = nestedMap
+      for (key in keyPath) {
+        Assertions.assertThat(value).isInstanceOf(MutableMap::class.java)
+        nestedMap = value as Map<String, Any>?
+        Assertions.assertThat(nestedMap).containsKey(key)
```
Comment (cortinico) [inline packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsTest.kt:241]:
```suggestion
        Assertions.assertThat(key in nestedMap)...
```

**Review by cortinico — CHANGES_REQUESTED** [review]
Thanks for taking the time to sending this over @giolaq 🙏 
You will have to rebase + apply the suggestions I recommended before we can merge this one.

**Review by cortinico — APPROVED** [review]
LGTM 👍

**Pull request closed without merging**